### PR TITLE
Renaming test engine to CodeComplianceTest_Engine

### LINKED ISSUE #107: Rename/Creation of CodeComplianceTest_Engine
Would like to propose renaming of the current `Test_Engine` to be more specifically be called `CodeComplianceTest_Engine`.
This will allow greater scope for expanding and implementing further tests across many of the other categories of testing we are targeting. See here https://github.com/BHoM/Test_Toolkit/issues/106 although the nomenclature here is not necessarily fixed and warrants review and final decisions.

See also latest commits to Interopability PR [here ](https://github.com/BHoM/Test_Toolkit/pull/6) - creating nice separation of responsibility and focus of code.

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### Issues addressed by this PR
Fixes #107 

 ### Test files

 ### Changelog

 ### Additional comments

### CODE REVIEW

**Comment by IsakNaslundBh** [comment]
@al-fisher would you mind chipping in on the namespace here?

File: CodeComplianceTest_Engine/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -27,11 +27,11 @@
 // General Information about an assembly is controlled through the following 
 // set of attributes. Change these attribute values to modify the information
 // associated with an assembly.
-[assembly: AssemblyTitle("Test_Engine")]
+[assembly: AssemblyTitle("Test.CodeCompliance_Engine")]
```
Comment (IsakNaslundBh) [inline CodeComplianceTest_Engine/Properties/AssemblyInfo.cs]: This looks wrong?

File: CodeComplianceTest_Engine/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -27,11 +27,11 @@
...
 [assembly: AssemblyDescription("")]
 [assembly: AssemblyConfiguration("")]
 [assembly: AssemblyCompany("")]
-[assembly: AssemblyProduct("Test_Engine")]
+[assembly: AssemblyProduct("Test.CodeCompliance_Engine")]
```
Comment (IsakNaslundBh) [inline CodeComplianceTest_Engine/Properties/AssemblyInfo.cs]: Same here

File: CodeComplianceTest_Test/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -5,12 +5,12 @@
 // General Information about an assembly is controlled through the following 
 // set of attributes. Change these attribute values to modify the information
 // associated with an assembly.
-[assembly: AssemblyTitle("Test_Test")]
+[assembly: AssemblyTitle("Test.CodeCompliance_Test")]
```
Comment (IsakNaslundBh) [inline CodeComplianceTest_Test/Properties/AssemblyInfo.cs]: `CodeComplianceTest_Test`

File: CodeComplianceTest_Test/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -5,12 +5,12 @@
...
 [assembly: AssemblyDescription("")]
 [assembly: AssemblyConfiguration("")]
 [assembly: AssemblyCompany("")]
-[assembly: AssemblyProduct("Test_Test")]
-[assembly: AssemblyCopyright("Copyright ©  2019")]
+[assembly: AssemblyProduct("Test.CodeCompliance_Test")]
```
Comment (IsakNaslundBh) [inline CodeComplianceTest_Test/Properties/AssemblyInfo.cs]: `CodeComplianceTest_Test`

**Comment by al-fisher** [comment]
I like:

``` C# 
namespace BH.Engine.Test.CodeCompliance
``` 
``` C# 
namespace BH.Engine.Test.Interoperability
```

👍

Allows for things like 
``` C# 
namespace BH.Engine.Test.Performance
``` 
``` C# 
namespace BH.Engine.Test.Installer
``` 
etc.

**Pull request merged**